### PR TITLE
MEN-4420: Add mender-configure git recipe

### PR DESCRIPTION
**Pull request by lluiscampos**

### CODE REVIEW

**Comment by lluiscampos** [comment]
Goes together with:
* https://github.com/mendersoftware/mender-image-tests/pull/39
* https://github.com/mendersoftware/mender-qa/pull/479

Testing at [Pipeline 252020495](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/252020495).

**Comment by lluiscampos** [comment]
New build @ [Pipeline 252053513](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/252053513)

File: meta-mender-core/recipes-mender/mender-configure/mender-configure.inc [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+RDEPENDS_${PN} = "dbus curl mender-client (>= 2.5)"
+
+FILES_${PN} = " \
+    ${datadir}/mender/modules/v3/mender-configure \
+    ${datadir}/mender/inventory/mender-inventory-mender-configure \
+"
+
+S = "${WORKDIR}/git"
+
+do_install() {
```
Comment (kacf) [inline meta-mender-core/recipes-mender/mender-configure/mender-configure.inc:13]: Propose to also add:

```
do_install_append_mender-image() {
    install -d -m 755 ${D}/data/mender-configure
    ln -s /data/mender-configure ${D}/var/lib/mender-configure
}
```

File: meta-mender-core/recipes-mender/mender-configure/mender-configure_git.bb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+SRC_URI = "git://github.com/mendersoftware/mender-configure-module.git;protocol=https;branch=${MENDER_CONFIGURE_BRANCH}"
+
+# DO NOT change the checksum here without make sure that ALL licenses (including
+# dependencies) are included in the LICENSE variable below. Note that for
+# releases, we must check the LIC_FILES_CHKSUM.sha256 file, not the LICENSE
+# file.
+def mender_configure_license(branch):
+    # Only one currently. If the sub licenses change we may introduce more.
+    return {
+               "license": "Apache-2.0 & BSD-2-Clause & BSD-3-Clause & ISC & MIT",
```
Comment (kacf) [inline meta-mender-core/recipes-mender/mender-configure/mender-configure_git.bb:59]: Mender-configure is Apache-2.0 only.

**Review by kacf — CHANGES_REQUESTED** [review]
Nice and clean!

File: meta-mender-core/recipes-mender/mender-configure/mender-configure.inc (hunk 1 above)
Comment (lluiscampos) [reply meta-mender-core/recipes-mender/mender-configure/mender-configure.inc:13]: Right. I'll add it and modify the tests accordingly.

File: meta-mender-core/recipes-mender/mender-configure/mender-configure_git.bb (hunk 2 above)
Comment (lluiscampos) [reply meta-mender-core/recipes-mender/mender-configure/mender-configure_git.bb:59]: I just copied it from mender-connect. Thank you for paying attention.

**Comment by lluiscampos** [comment]
@kacf Updated both this PR and https://github.com/mendersoftware/mender-image-tests/pull/39

**Comment by lluiscampos** [comment]
New build @ [Pipeline 252100822](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/252100822)

**Comment by lluiscampos** [comment]
New build @ [Pipeline 252128984](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/252128984)

**Comment by lluiscampos** [comment]
@kacf Please have a look at the last commit. I think fixing the test is the right thing to do (instead of modifying the behavior of `modules` in `PACKAGECONFIG`), but let me know if you think otherwise.

**Comment by lluiscampos** [comment]
> Hello smile_cat I created a pipeline for you here: [Pipeline-252433835](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/252433835)
> Build Configuration Matrix

Now that I realized that this PR does not strictly depend on https://github.com/mendersoftware/mender-qa/pull/479, I will leave the default build to run (it should pass just fine).

**Review by kacf — APPROVED** [review]
> Please have a look at the last commit. I think fixing the test is the right thing to do (instead of modifying the behavior of `modules` in `PACKAGECONFIG`), but let me know if you think otherwise.

Yes, fully agree!

**Pull request merged**